### PR TITLE
Config UI page

### PR DESCRIPTION
**Pull request by Jespertastesen**
This may help or make it easier for people to install the gallery the first time.

This contans:
- Update to newest Bootstrap 4.3.1
- Settings convertet to json - can be updated from UI
-> no changes made to the way the settings is read form $config
- Show/hide secret key
- Generate new from the UI
- Minor code cleaning of whitespace and tab sizes.

settings UI can be accessed from `domain.com/u/setup` or by link in the mail gallery at the bottom.

Missing:

- Security
-> remove http access to settings.json to make it more secure, and prevent others from just downloading the file.
- Update "How to"

Annotation 2019-11-20 195611

### CODE REVIEW

File: src/u/js/main.js [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
-$('[data-toggle="tooltip"]').tooltip();
\ No newline at end of file
+
+
+
+});
+$('[data-toggle="tooltip"]').tooltip();
+
+
+function makeid() {
```
Comment (JoeGandy) [inline src/u/js/main.js]: Encase we need this useful function again, have a parameter for the id length

File: src/u/setup/index.php [hunk 2]
```diff
@@ -0,0 +1,273 @@
+<?php
+if (isset($_POST['save'])) {
+
+    $settings = new \stdClass;
```
Comment (JoeGandy) [inline src/u/setup/index.php]: Probably better as an associative array

Comment (Jespertastesen) [reply src/u/setup/index.php]: the obj, is just to make it not throw an error.
it outputs as an keyed array, that generated the json file.

File: src/u/js/main.js (hunk 1 above)
Comment (Jespertastesen) [reply src/u/js/main.js]: this has been added

**Comment by Jespertastesen** [comment]
Some issues fixed in #66 

in general, there isnt really a reason for users to "upgrade" from their current setup if it works.
but for newcommers it should be easier to setup.

I have been thinking about making a "first setup guide" in style with software installation.
Here the user will get promtet with the required fields to be filled in a nice looking UI.

Annotation 2019-11-21 230220

**Comment by JoeGandy** [comment]
Sounds great, I'd prob keep it as a seperate PR once this one is in,
Speaking of which the new changes look good, you may need to re create the PR so its not already merged

**Comment by Jespertastesen** [comment]
Ofc that would be in a seperate PR.

Was thinking about going the SQLite way to store the info + make a signin option.

next PR @JoeGandy

RFYiSblWh3

**Pull request merged**